server/api/datasets/[datasetid]/agreements.get.ts

```diff
@@ -0,0 +1,92 @@
+import type { QueryValue } from "ufo";
+
+/**
+ * Attempts to parse the value of the given query parameter to a numeric value.
+ *
+ * @param param query parameter value
+ * @param defaultValue default value to return if the parameter cannot be parsed
+ * @returns the parsed numeric value of the parameter, or the default value
+ */
+function parseNumericParamWithDefault(
+  param: QueryValue,
+  defaultValue: number,
+): number {
+  let parsedValue = NaN;
+
+  if (typeof param === "string") {
+    parsedValue = Number.parseInt(param);
+  }
+
+  return Number.isFinite(parsedValue) ? parsedValue : defaultValue;
+}
+
+export default defineEventHandler(async (event) => {
+  const datasetId = event.context.params?.datasetid;
+  const query = getQuery(event);
+  const page = parseNumericParamWithDefault(query.page, 1);
+  const pageSize = parseNumericParamWithDefault(query.pageSize, 10);
+
+  if (!datasetId) {
+    throw createError({
+      statusCode: 400,
+      statusMessage: "Dataset ID parameter is required",
+    });
+  }
+
+  if (page <= 0) {
+    throw createError({
+      statusCode: 400,
+      statusMessage: `Page parameter must be greater than 0, but was ${page}`,
+    });
+  }
+
+  if (pageSize <= 0) {
+    throw createError({
+      statusCode: 400,
+      statusMessage: `Page size parameter must be greater than 0, but was ${pageSize}`,
+    });
+  }
+
+  const whereClause = {
+    dataset_id: datasetId,
```

Comment:
I've already had some feedback from Shannon regarding which ones we'll want to show, particularly since we are getting requests that we won't necessarily fulfill since they are from places where the we aren't allowed to export data. We will want to add another parameter to limit based on a particular request status, but we have to first have that new status so this is all good at present.